### PR TITLE
Switch to errorprone-javacplugin

### LINKED ISSUE #218: Move to new error prone plugin
https://github.com/tbroyer/gradle-errorprone-javacplugin-plugin

Better compatibility with gradle moving forward, though not sure why the original wasn't just updated with these changes 🤔 

FYI @msridhar @kageiit

### PR DESCRIPTION
**Pull request by hzsweers**
Also opportunistically updates ErrorProne and NullAway to latest, as well as enabling nullaway on the static analysis artifact

Resolves #218

### CODE REVIEW

**Comment by hzsweers** [comment]
Forgot the JDK 9 requirement. Does that make sense to require for building or should we hold off?

This would definitely bork anyone in uber trying to build this since our android env mandates jdk 8 :|

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ android:
     - platform-tools
 
 jdk:
-  - oraclejdk8
+  - oraclejdk9
```
Comment (kageiit) [inline .travis.yml]: 😮

Comment (kageiit) [reply .travis.yml]: you sure you want to change this?

Comment (hzsweers) [reply .travis.yml]: Well, that's why I asked. Also it looks like we'd need to go to 10, 9 isn't even downloadable anymore

**Comment by hzsweers** [comment]
Seems android can't even build with JDK 10, so going to shutter this for now

**Comment by hzsweers** [comment]
Will do the EP/Nullaway updates and enabling nullaway in the EP checker separately

**Pull request closed without merging**